internal/states/encryption/encryptionconfig/source.go

```diff
@@ -0,0 +1,15 @@
+package encryptionconfig
+
+type Source string
+
+const (
+	// SourceHCL indicates that the configuration has been obtained by parsing the HCL source.
+	SourceHCL Source = "hcl"
```

Comment:
Food for thought: We don't just support HCL, we also support JSON configuration format (https://opentofu.org/docs/language/syntax/json/).

Do we need another way to define that it came from the config files and not from the env?

Reply:
Or possibly change the source to "config" from "hcl"?